### PR TITLE
Update TODO app to latest UI5 recommendations

### PR DESCRIPTION
**Pull request by Michadelic**
* CSP Compliant entry points
* Declarative Component usage
* Updated OPA & Unit test setup
* Removed jQuery
* Added . in view events
* Minor cosmetic code cleanup

### CODE REVIEW

File: webapp/Component.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-sap.ui.define(["sap/ui/core/UIComponent"], function(UIComponent){
+sap.ui.define(["sap/ui/core/UIComponent", "sap/ui/core/ComponentSupport"], function(UIComponent) {
```
Comment (Michadelic) [inline webapp/Component.js]: shouldn't it be identified by the bootstrap tag automatically? i'm preety sure app developers will forget to load this dependency frequently if it has to be specified again here in the Component

Comment (codeworrior) [reply webapp/Component.js]: it should, but it isn't. I think Matthias added the dependency as a workaround.

Comment (matz3) [reply webapp/Component.js]: Exactly. See https://github.com/SAP/ui5-builder/issues/92

Comment (Michadelic) [reply webapp/Component.js]: thanks for clarifying, fine for me

File: webapp/index-self-contained.html [hunk 2]
```diff
@@ -1,42 +1,23 @@
...
-	<meta http-equiv="X-UA-Compatible" content="IE=edge" >
-	<meta charset="UTF-8">
-
 	<script id="sap-ui-bootstrap"
 		src="resources/sap-ui-custom.js"
-		data-sap-ui-theme="sap_belize"
-		data-sap-ui-libs="sap.m"
-		data-sap-ui-bindingSyntax="complex"
+		data-sap-ui-theme="sap_belize_plus"
+		data-sap-ui-resourceroots='{
```
[the remark this answers is not in the document]
Comment (matz3) [reply webapp/index-self-contained.html]: added the first approach 👍

**Pull request merged**